tests/mr/device/mr_ref_test.hpp

```diff
@@ -71,17 +71,20 @@ inline void test_get_current_device_resource()
   void* ptr = rmm::mr::get_current_device_resource()->allocate(1_MiB);
   EXPECT_NE(nullptr, ptr);
   EXPECT_TRUE(is_properly_aligned(ptr));
-  EXPECT_TRUE(is_device_accessible_memory(ptr));
+  if (dynamic_cast<rmm::mr::system_memory_resource*>(rmm::mr::get_current_device_resource()) ==
+      nullptr) {
+    EXPECT_TRUE(is_device_accessible_memory(ptr));
+  }
   rmm::mr::get_current_device_resource()->deallocate(ptr, 1_MiB);
 }
 
-inline void test_allocate(resource_ref ref, std::size_t bytes)
+inline void test_allocate(resource_ref ref, std::size_t bytes, bool is_system_mr)
 {
   try {
     void* ptr = ref.allocate(bytes);
     EXPECT_NE(nullptr, ptr);
     EXPECT_TRUE(is_properly_aligned(ptr));
-    EXPECT_TRUE(is_device_accessible_memory(ptr));
+    if (not is_system_mr) { EXPECT_TRUE(is_device_accessible_memory(ptr)); }
```

Comment:
If it's not device accessible memory, you shouldn't even be running these tests on it. If it's device accessible but `is_device_accessible_memory` returns false, then we should investigate why it returns false and fix it.  I don't really like the extra boolean parameter, the allocation tests should be generic.

Reply:
Done.